### PR TITLE
Throw `ResourceNotFoundException` outside of re-defined error-handler

### PR DESCRIPTION
**Pull request by dmitry-varennikov-eventbase**
Throwing exception outside of re-defined error-handler allows us to catch it in our app stack

### CODE REVIEW

**Comment by Maks3w** [comment]
Please avoid all those formatting changes, are noisy,

Provide a unit test.

File: src/JsonSchema/Uri/Retrievers/FileGetContents.php [hunk 1]
```diff
@@ -14,51 +14,73 @@
...
+                $errline,
+                $errcontext,
+            ];
         });
         $response = file_get_contents($uri);
         restore_error_handler();
 
+        if ($error) {
+            list (, $errstr, $errfile, $errline) = $error;
+            $message = sprintf('%s on line %s in %s', $errstr, $errline,
```
Comment (Maks3w) [inline src/JsonSchema/Uri/Retrievers/FileGetContents.php]: Almost everything in the proposed message template is already present in the stacktrace. $errstr is the only value you need here.

File: src/JsonSchema/Uri/Retrievers/FileGetContents.php [hunk 2]
```diff
@@ -14,51 +14,73 @@
...
+                $errfile,
+                $errline,
+                $errcontext,
+            ];
         });
         $response = file_get_contents($uri);
         restore_error_handler();
 
+        if ($error) {
+            list (, $errstr, $errfile, $errline) = $error;
```
Comment (Maks3w) [inline src/JsonSchema/Uri/Retrievers/FileGetContents.php]: Avoid discard values, just store what you really need in $error.

**Comment by dmitry-varennikov-eventbase** [comment]
Unit test `JsonSchema\Tests\Uri\Retrievers\FileGetContentsTest::testFetchMissingFile`

**Comment by bighappyface** [comment]
@Maks3w https://github.com/justinrainbow/json-schema/pull/273/files?w=1 to exclude white space

**Comment by Maks3w** [comment]
This is not about visualization. It's about to keep a clean and organized history. 

If the CS changes was done in a different PR you didn't have to revert.

**Comment by bighappyface** [comment]
@Maks3w I'm starting to question the value of considering your input when you basically have nothing good or nice to say about anything and have contributed little to this project but hyper-criticism.

How about you https://m.spreadshirt.com/shut-the-fuck-up-and-write-some-code-t-shirts-A103693056

**Comment by jojo1981** [comment]
@Maks3w What about the "The Boy Scout Rule"? Why you're complaining about this nice PR?

**Comment by dmitry-varennikov-eventbase** [comment]
Does this PR require any more code?
Is this PR could be merged any time soon?

**Comment by bighappyface** [comment]
+1

**Comment by bighappyface** [comment]
Once this is merged does anyone see any issue with bumping the package to 2.0.5?

**Pull request merged**